src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1618,23 +1618,24 @@ public static IEnumerable<object[]> Join_ObjectArray_TestData()
             yield return new object[] { null, new object[] { "Foo", "Bar", "Baz" }, "FooBarBaz" };
             yield return new object[] { "$$", new object[] { "Foo", null, "Baz" }, "Foo$$$$Baz" };
 
-            // Join does nothing if array[0] is null
-            yield return new object[] { "$$", new object[] { null, "Bar", "Baz" }, "" };
+            // Test join when first value is null
+            yield return new object[] { "$$", new object[] { null, "Bar", "Baz" }, "$$Bar$$Baz" };
```

Comment:
I think it pass test on NetFramework because I wrote next code
```cs
#if !netstandard17
            if (values.Length > 0 && values[0] == null) // Join return nothing when first value is null
                expected = "";
#endif
```